Documentation/learning/glossary.md

```diff
@@ -26,10 +30,93 @@ Peer is another member of the same cluster.
 
 A proposal is a request (for example a write request, a configuration change request) that needs to go through raft protocol.
 
+## Quorum
+
+The number of members of a group/cluster required to be present to perform a set of instructions/operations, usually a majority.
+
 ## Client
 
 Client is a caller of the cluster's HTTP API.
 
-## Machine (deprecated)
+## Command
 
-The alternative of Member in etcd before 2.0
+Command is an identifiers which are used to perform a specific operation in etcd server using etcdctl.
+Example: put, get, del, lease, etc.
+
+## Transaction
+
+Transaction is a set of instructions which are to be performed as an atomic operation based on a certain set of conditions.
+
+## Store
+
+Store is a place in etcd server which stores key-value pairs.
+
+## Keys
+
+Keys are unique idetifiers. Keys are used to store values to the store and the values can be retrieved from the store using the keys.
+In certain cases like server certification a key can be referred to as a unique identifier used for encrypting the values.
+
+## Keyspaces
+
+Keyspaces refer to the set of all possible permutations of a key.
+
+## Lease
+
+Lease is a contract given for a specified time. It can be used to attach to the keys to make keys available as long as the lease lasts. Once the lease expires all the keys associated with it also expire.
+
+## Revision
+
+Revision is the current revision of etcd. It is incremented every time the v3 backed is modified (e.g., Put, Delete, Txn).
+
+## ModRevision
+
+ModRevision is the etcd revision of the last update done on a key. Every time a key gets updated the modrevision gets incremented by 1.
+
+## Version
```

Comment:
Yes. I want to make them separate. Actually I think software version is not etcd specific. i do not think we need to mention it.